### PR TITLE
Antony/prov rep

### PR DESCRIPTION
**Pull request by antony-wilson**
The prov report is now based on the DataProduct

This branch includes the changes made in the [nathan/prov-report](https://github.com/FAIRDataPipeline/data-registry/tree/nathan/prov-report)

### CODE REVIEW

File: data_management/prov.py [hunk 1]
```diff
@@ -1,133 +1,356 @@
...
 
     This uses the W3C PROV ontology (https://www.w3.org/TR/prov-o/).
 
-    :param code_run: The CodeRun to generate the PROV document for
+    :param data_product: The DataProduct to generate the PROV document for
+
     :return: A PROV-O document
+
     """
+    url = ''.join(['http://', get_current_site(None).domain, '/'])
```
Comment (alahiff) [inline data_management/prov.py:249]: This isn't correct in general, e.g. for the remote registry we use `https`. Also, when testing this branch in a local registry it gives the wrong site name (for me at least):
```
    "prefix": {
        "default": "http://example.com/"
    },
```
One way to fix this problem, which I've tested, is to replace this line with:
```
url = request.build_absolute_uri('/')
```
 For this to work, in the `get` method in `ProvReportView` (in `data_management/rest/views.py`) you need to pass `request` into `generate_prov_document`.

**Comment by alahiff** [comment]
If graphviz is not installed, when you try to look at a prov report you get an internal server error due to:
```
Exception Type: FileNotFoundError at /api/prov-report/1/
Exception Value: [Errno 2] "dot" not found in path.
```
It would be good if we could give a nicer error than a 500, and perhaps when the local registry is installed we could check for the existence of `dot` and if it's not there give a warning message.

**Comment by richardreeve** [comment]
@alahiff Antony is on holiday for the next week now unfortunately, so if there's any way you or @nathan-cummings-ukaea could make these edits it would be incredibly helpful!

**Comment by alahiff** [comment]
I've added a commit (https://github.com/FAIRDataPipeline/data-registry/pull/128/commits/0cfaae83054a1e44fef894b482999f5efe694806) to fix the issue with the URL being incorrect

**Comment by richardreeve** [comment]
Thanks @alahiff! I've created a new issue - #130 - about the internal server error. Apart from that, is this good to merge? I notice some checks are failing - is this a known problem?

File: data_management/tests/test_api.py [hunk 2]
```diff
@@ -789,3 +790,341 @@ def test_filter_by_key(self):
...
+        )
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(
+            response["Content-Type"], "text/provenance-notation; charset=utf8"
+        )
+
+        result_bits = response.data.split("last_updated")
+        result_end = result_bits[1].split("xsd:dateTime, ", 1)[1]
+        result = result_bits[0] + result_end
+        expected_result = """document
```
Comment (alahiff) [inline data_management/tests/test_api.py:1080]: This is no longer valid, as it contains a hardwired URL involving https://data.scrc.uk/

Comment (alahiff) [reply data_management/tests/test_api.py:1080]: https://github.com/FAIRDataPipeline/data-registry/pull/128/commits/dead28c0e3c3913e5fa1a17cf0340b3b9b0208bc

**Comment by richardreeve** [comment]
Is this good to merge then @alahiff?

**Pull request merged**